### PR TITLE
ENH: q2cli import & extract API cleanup and minor typo fix

### LINKED ISSUE #190: Typos in `tools export` help text
```bash
qiime tools export --help                                                                                                                                                                                                                                                                                  
Usage: qiime tools export [OPTIONS]

  Exporting extracts (and optionally transforms) datastored inside an
  Artifact or Visualization. Note thatVisualizations cannot be transformed
  with --output-format

Options:
  --input-path PATH     [required]
  --output-path PATH    Directory where data should be exported to  [required]
  --output-format TEXT  Format which the data should be exported as. --output-
                        format cannot be used with Visualizations
  --help                Show this message and exit.
```

---

> datastored

This should read as `data stored`

---

> thatVisualizations

This should read as `that Visualizations`.

### LINKED ISSUE #191: API Change: update `tools import` to match input/output naming convention
Part of the changes introduced in #189 were some breaking changes w.r.t. the naming convention used for the parm flags in `tools export`.

Old:

```bash
qiime tools export --help                                                                                                                                                                                                                                                                                  
Usage: qiime tools export [OPTIONS] PATH

  Export data from a QIIME 2 Artifact or Visualization. Exporting extracts
  the data stored in an Artifact or Visualization and will support exporting
  to multiple formats in the future. For now, the data is exported in the
  format it is stored within the Artifact or Visualization. Use 'qiime tools
  extract' to extract the Artifact or Visualization's entire archive.

Options:
  --output-dir DIRECTORY  Directory where data should be exported to
                          [required]
  --help                  Show this message and exit.

```

New:

```bash
qiime tools export --help                                                                                                                                                                                                                                                                                   master 517edc7
Usage: qiime tools export [OPTIONS]

  Exporting extracts (and optionally transforms) datastored inside an
  Artifact or Visualization. Note thatVisualizations cannot be transformed
  with --output-format

Options:
  --input-path PATH     [required]
  --output-path PATH    Directory where data should be exported to  [required]
  --output-format TEXT  Format which the data should be exported as. --output-
                        format cannot be used with Visualizations
  --help                Show this message and exit.

```

Note the changes in `PATH` as a positional param vs `--input-path` as  a flag, as well as the renaming of `--output-dir` to `--output-path`.

---

Regarding `tools import`, we would like to see similar changes implemented there, so that there is symmetry between the two commands.

Current:

```bash
qiime tools import --help                                                                                                                                                                                                                                                                                  
Usage: qiime tools import [OPTIONS]

  Import data to create a new QIIME 2 Artifact. See https://docs.qiime2.org/
  for usage examples and details on the file types and associated semantic
  types that can be imported.

Options:
  --type TEXT                The semantic type of the artifact that will be
                             created upon importing. Use --show-importable-
                             types to see what importable semantic types are
                             available in the current deployment.  [required]
  --input-path PATH          Path to file or directory that should be
                             imported.  [required]
  --output-path PATH         Path where output artifact should be written.
                             [required]
  --source-format TEXT       The format of the data to be imported. If not
                             provided, data must be in the format expected by
                             the semantic type provided via --type.
  --show-importable-types    Show the semantic types that can be supplied to
                             --type to import data into an artifact.
  --show-importable-formats  Show formats that can be supplied to --source-
                             format to import data into an artifact.
  --help                     Show this message and exit.
```

Proposed:

```bash
qiime tools import --help                                                                                                                                                                                                                                                                                  
Usage: qiime tools import [OPTIONS]

  Import data to create a new QIIME 2 Artifact. See https://docs.qiime2.org/
  for usage examples and details on the file types and associated semantic
  types that can be imported.

Options:
  --type TEXT                The semantic type of the artifact that will be
                             created upon importing. Use --show-importable-
                             types to see what importable semantic types are
                             available in the current deployment.  [required]
  --input-path PATH          Path to file or directory that should be
                             imported.  [required]
  --output-path PATH         Path where output artifact should be written.
                             [required]
  --input-format TEXT        The format of the data to be imported. If not
                             provided, data must be in the format expected by
                             the semantic type provided via --type.
  --show-importable-types    Show the semantic types that can be supplied to
                             --type to import data into an artifact.
  --show-importable-formats  Show formats that can be supplied to --source-
                             format to import data into an artifact.
  --help                     Show this message and exit.
```

Note that change from `--source-format` to `--input-format`.

This change will likely impact the user docs (we will open another issue there if needed).

This discussion recently came up here: https://github.com/qiime2/q2cli/pull/189#issuecomment-398581368.

### LINKED ISSUE #194: API Change: update `tools extract` to match input/output naming convention
Related to #191 .

---

## Current behavior

```bash
qiime tools extract --help
Usage: qiime tools extract [OPTIONS] PATH

  Extract all contents of a QIIME 2 Artifact or Visualization's archive,
  including provenance, metadata, and actual data. Use 'qiime tools export'
  to export only the data stored in an Artifact or Visualization, with the
  choice of exporting to different formats.

Options:
  --output-dir DIRECTORY  Directory where archive should be extracted to
                          [default: current working directory]
  --help                  Show this message and exit.
```

Similar to #191 - we should drop the positional `PATH` argument in favor for `--input-path`, and we should rename `--output-dir` to `--output-path`.

### PR DESCRIPTION
**Pull request by turanoo**
Fixes #190 
Fixes #191 
Fixes #194

### CODE REVIEW

File: q2cli/tools.py [hunk 1]
```diff
@@ -330,26 +330,29 @@ def view(visualization_path, index_extension):
...
                     "the data stored in an Artifact or Visualization, with "
                     "the choice of exporting to different formats.")
-@click.argument('path', type=click.Path(exists=True, file_okay=True,
-                                        dir_okay=False, readable=True))
-@q2cli.option('--output-dir', required=False,
-              type=click.Path(exists=True, file_okay=False, dir_okay=True,
+@q2cli.option('--input-path', required=True,
+              type=click.Path(exists=True, file_okay=True, dir_okay=False,
+                              readable=True),
+              help='Directory where archive should be extracted from')
```
Comment (thermokarst) [inline q2cli/tools.py]: How about we match the `input-path` text in `import`?

`Path to file that should be extracted`

File: q2cli/tools.py [hunk 2]
```diff
@@ -19,10 +19,10 @@ def tools():
...
-               short_help='Export data from a QIIME 2 Artifact'
+               short_help='Export data from a QIIME 2 Artifact '
                'or a Visualization',
-               help='Exporting extracts (and optionally transforms) data'
-               'stored inside an Artifact or Visualization. Note that'
+               help='Exporting extracts (and optionally transforms) data '
+               'stored inside an Artifact or Visualization. Note that '
                'Visualizations cannot be transformed with --output-format'
                )
 @q2cli.option('--input-path', required=True,
```
Comment (thermokarst) [inline q2cli/tools.py]: Can we add a description to this option?

How about we match the `input-path` text in `import`?

`Path to file that should be exported`

File: q2cli/tools.py (hunk 1 above)
Comment (turanoo) [reply q2cli/tools.py]: Yeah, I think that sounds better!

File: q2cli/tools.py [hunk 3]
```diff
@@ -19,21 +19,23 @@ def tools():
...
+              help='Path to file that should be exported')
 @q2cli.option('--output-path', required=True,
               type=click.Path(exists=False, file_okay=True, dir_okay=True,
                               writable=True),
-              help='Directory where data should be exported to')
+              help='Path to file or directory where '
+              'data should be exported to')
 @q2cli.option('--output-format', required=False,
-              help='Format which the data should be exported as. '
+              help='Format which the data should be exported as. \n'
```
Comment (turanoo) [inline q2cli/tools.py]: --output-format was being broken down into two words when shown on the command line. I added a new line to prevent this!

Comment (thermokarst) [reply q2cli/tools.py]: Good catch! How about instead we say `Format which [...]. This option cannot be used with Visualizations.`?

Comment (turanoo) [reply q2cli/tools.py]: I like that! I will change it now ⚙️

**Comment by thermokarst** [comment]
Thanks @turanoo! Please update the [changelog](https://forum.qiime2.org/t/preview-qiime-2-2018-8-development-changelog/4724?u=thermokarst) when you get a minute. It is worth mentioning in the q2cli section of the changelog that there are several breaking changes to the CLI API, and that users should update their scripts and workflows accordingly. Maybe one way of doing that is to demonstrate what the old command looks like compared to the new command. Let us know if you need help. Thanks!

**Pull request merged**